craft_parts/lifecycle_manager.py

```diff
@@ -40,11 +41,13 @@ class LifecycleManager:
         to the :ref:`parts schema<parts-schema>`. The format is compatible with the
         output generated by PyYAML's ``yaml.load``.
     :param application_name: A unique non-empty identifier for the application
-        using Craft Parts. This string will be used as segregated directory path
-        when creating persistent data that shouldn't be shared with other
-        applications and as a prefix to step environment variable names. Valid
-        application names contain upper and lower case letters, underscores
-        or numbers, and must start with a letter.
+        using Craft Parts. The application name is used as a prefix to step
+        environment variables, and as a component of the cache directory path if
+        it's not set by the application. Valid application names contain upper
```

Comment:
the fall back is nice for backwards compatibility cases, but we don't have that. I would personally just remove it.

Reply:
Removed.